receiver/LoraToGPSServer/device/device.go

```diff
@@ -322,31 +332,174 @@ func irnasParseSingle(data dataInterface) (*Data, error) {
 	return dataParsed, nil
 }
 
+
+
+
+func G62Parse(data *DataUpPayload) ([]*Data, error) {
+
+	dataParsed := &Data{
+		Valid: true,
+		Attr:  map[string]string{},
+	}
+
+	// Non GPS data.
+	if data.FPort != 1 {
+		dataParsed.Valid = false
+		if os.Getenv("DEBUG") == "1" {
+			log.Printf("skipping non gps data, fport:%+v", data.FPort)
+		}
+		return []*Data{dataParsed}, nil
+	}
+
+	lat, ok := data.Object["latitudeDeg"]
+	if !ok {
+		return nil, errors.New("data doesn't contain lat")
+	}
+	lon, ok := data.Object["longitudeDeg"]
+	
+	if !ok {
+		return nil, errors.New("data object doesn't contain lon")
+	}
+	//data.DevEUI=data.DeviceInfo.DevEui
+
+	dataParsed.Time = int64( data.Time.Unix())
+	
+
+	// Port 12 status messages contain only lat/lon.
+	hdop, ok := data.Object["hdop"]
+	if !ok {
+		log.Printf("data object doesn't contain hdop so setting to 0")
+		hdop = 0.0
+	}
+	dataParsed.Hdop = hdop.(float64)
+
+	dataParsed.Lat = lat.(float64)
+	dataParsed.Lon = lon.(float64)
+	if dataParsed.Lat == 0.0 || dataParsed.Lon == 0.0 {
+		dataParsed.Valid = false
+	}
+	/*if val, ok := data["gps_time"]; ok { // From system updates.
+		dataParsed.Time = int64(val.(float64))
+	}
+	if val, ok := data["time"]; ok { // From periodic or motion triggered updates.
+		dataParsed.Time = int64(val.(float64))
+	}*/
+
+	if val, ok := data.Object["speedKmph"]; ok {
+		dataParsed.Attr["speedKmph"] = fmt.Sprintf("%v", val.(float64))
+	}
```

Comment:
```suggestion
	if val, ok := data.Object["speedKmph"]; ok {
		dataParsed.Speed = fmt.Sprintf("%v", val.(float64))
		dataParsed.Motion = true
	}
```

instead of this should add it to `dataParsed.Speed` and then in the `Parse` function can change it to
```
if point.Motion && point.Speed == 0  {
	point.Speed = self.Speed(point.ID)
}
```

Reply:
I don't see any code changes here.